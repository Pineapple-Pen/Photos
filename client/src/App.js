import React from 'react';
import ReactDOM from 'react-dom';
import App from './components/App';
import '../dist/styles.css';

//if window exists append it otherwise export
if (window){
  window.App = App;
  ReactDOM.render(<App />, document.getElementById('gallery-app'));

} 

export default App;
