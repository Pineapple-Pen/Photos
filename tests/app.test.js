import React from 'react';
import axios from 'axios';
import MockAdapter from 'axios-mock-adapter';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import App from '../client/src/components/App';
import testData from './test_data';

configure({ adapter: new Adapter() });

describe('Photo Gallery App', () => {
  const mock = new MockAdapter(axios);
  const data = testData;

  mock.onGet('/api/restaurants/:id').reply(200, data);

  it('should render correctly', () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toMatchSnapshot();
  });
});
