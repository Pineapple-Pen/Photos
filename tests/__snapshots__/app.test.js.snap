// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Photo Gallery App should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="gallery"
        >
          <TopNav />
          <Gallery
            columns={4}
            margin={3}
            onClick={[Function]}
            photos={Array []}
          />
          <div
            className="photo-counter"
            onClick={[Function]}
            role="presentation"
          >
            0
             PHOTOS +
          </div>
        </div>,
        <SlideShowView
          className="slideshow"
          clickNext={[Function]}
          clickPrev={[Function]}
          closeLightbox={[Function]}
          current={0}
          gridButtonClick={[Function]}
          isLightboxOpen={false}
          photos={Array []}
          placeName=""
        />,
      ],
      "id": "main-app",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <TopNav />,
            <Gallery
              columns={4}
              margin={3}
              onClick={[Function]}
              photos={Array []}
            />,
            <div
              className="photo-counter"
              onClick={[Function]}
              role="presentation"
            >
              0
               PHOTOS +
            </div>,
          ],
          "className": "gallery",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "columns": 4,
              "margin": 3,
              "onClick": [Function],
              "photos": Array [],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                0,
                " PHOTOS +",
              ],
              "className": "photo-counter",
              "onClick": [Function],
              "role": "presentation",
            },
            "ref": null,
            "rendered": Array [
              0,
              " PHOTOS +",
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "className": "slideshow",
          "clickNext": [Function],
          "clickPrev": [Function],
          "closeLightbox": [Function],
          "current": 0,
          "gridButtonClick": [Function],
          "isLightboxOpen": false,
          "photos": Array [],
          "placeName": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="gallery"
          >
            <TopNav />
            <Gallery
              columns={4}
              margin={3}
              onClick={[Function]}
              photos={Array []}
            />
            <div
              className="photo-counter"
              onClick={[Function]}
              role="presentation"
            >
              0
               PHOTOS +
            </div>
          </div>,
          <SlideShowView
            className="slideshow"
            clickNext={[Function]}
            clickPrev={[Function]}
            closeLightbox={[Function]}
            current={0}
            gridButtonClick={[Function]}
            isLightboxOpen={false}
            photos={Array []}
            placeName=""
          />,
        ],
        "id": "main-app",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <TopNav />,
              <Gallery
                columns={4}
                margin={3}
                onClick={[Function]}
                photos={Array []}
              />,
              <div
                className="photo-counter"
                onClick={[Function]}
                role="presentation"
              >
                0
                 PHOTOS +
              </div>,
            ],
            "className": "gallery",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "columns": 4,
                "margin": 3,
                "onClick": [Function],
                "photos": Array [],
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  0,
                  " PHOTOS +",
                ],
                "className": "photo-counter",
                "onClick": [Function],
                "role": "presentation",
              },
              "ref": null,
              "rendered": Array [
                0,
                " PHOTOS +",
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "className": "slideshow",
            "clickNext": [Function],
            "clickPrev": [Function],
            "closeLightbox": [Function],
            "current": 0,
            "gridButtonClick": [Function],
            "isLightboxOpen": false,
            "photos": Array [],
            "placeName": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
